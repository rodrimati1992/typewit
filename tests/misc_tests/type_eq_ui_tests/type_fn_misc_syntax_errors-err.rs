typewit::type_fn!{
    struct Unit;
}

typewit::type_fn!{
    u8 => u16;
}

fn main(){}