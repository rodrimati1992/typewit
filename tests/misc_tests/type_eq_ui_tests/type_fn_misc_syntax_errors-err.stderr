error: expected type-level function definitions
 --> tests/misc_tests/type_eq_ui_tests/type_fn_misc_syntax_errors-err.rs:1:1
  |
1 | / typewit::type_fn!{
2 | |     struct Unit;
3 | | }
  | |_^
  |
  = note: this error originates in the macro `$crate::__tyfn_parse_fns` which comes from the expansion of the macro `typewit::type_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error: invalid argument for `type_fn` macro
       expected struct declaration followed by type-level function definitions
 --> tests/misc_tests/type_eq_ui_tests/type_fn_misc_syntax_errors-err.rs:5:1
  |
5 | / typewit::type_fn!{
6 | |     u8 => u16;
7 | | }
  | |_^
  |
  = note: this error originates in the macro `typewit::type_fn` (in Nightly builds, run with -Z macro-backtrace for more info)
